Minor edits to Initialization Options.md and updates to Instance Methods.md

Jeff17Robbins opened the pull request:
including adding `attachChild()` `detachChild`, removing some vestigial references to the removed `change` event, and attempting to document new `observe()` options.  

I left the `strict` option as a {{{TODO}}} because I don't know what it means.  

I also tried to add info about the `**` recursive wildcard.    

I didn't change `.observeOnce()`, pending feedback on my changes to `.observe()`, but I assume it too needs
updating.     

Finally, I don't really understand what `attachChild()` means when the child is *not* a component, but is merely a separate Ractive instance.  I think the doc needs more context about that.

evs-chris commented on docs/API/Instance Methods.md:
```diff
@@ -99,6 +99,48 @@ If an animation is started on a [keypath](../Concepts/Templates/Keypaths.md) whi
+
+- `child (Ractive instance)`: The child instance to attach.
+- `options (Object)`: Options to use when attaching the child:
+> > * **target** *`string`* - an anchor name at which to render the instance. See [`Components`](../Extend/Components.md). If the instance is already rendered, it will be unrendered and re-rendered at an appropriate anchor.
```
We should probably add an anchors section to the component docs and link directly to it.

evs-chris commented on docs/API/Instance Methods.md:
```diff
@@ -484,12 +548,16 @@ Observes the data at a particular [keypath](../Concepts/Templates/Keypaths.md).
     - `[init] (boolean)`: Defaults to `true`. Whether or not to initialise the observer, i.e. call the function with the current value of `keypath` as the first argument and `undefined` as the second.
     - `[defer] (boolean)`: Defaults to `false`, in which case [observers](../Concepts/Events/Publish-Subscribe.md) will fire before any DOM changes take place. If `true`, the [observer](../Concepts/Events/Publish-Subscribe.md) will fire once the DOM has been updated.
+    - `[links] (boolean)`: Defaults to `false`.  Whether or not the observer should "follow through" any links created with [ractive.link()](#link).
+    - `[strict] (boolean)`: Defaults to `false`. {{{TODO need explanation of this option}}}
```
`strict` uses object identity to determine if there was a change, meaning that unless the primary object changed, it won't trigger the observer. For instance with `{ data: { foo: { bar: 'baz' } } }`, `ractive.observe('foo', ..., { strict: true })` will not fire on `ractive.set('foo.bar', 'bat')` but will on `ractive.set('foo', { bar: 'bip' })`.

Jeff17Robbins replied:
Ok, I get that. What is the workflow to make an edit to the PR?  Ought I make the edit?  If so, is it an edit to the PR or do I need to make a new PR?    If an edit to the PR, how do I do that?  (Sorry for the newbie questions.)

fskreuz replied:
You can use the same branch you used for this PR. Make the edits, commit the changes, and push the branch up. Github will automatically add the commit in this PR.

evs-chris commented:
> Finally, I don't really understand what attachChild() means when the child is not a component, but is merely a separate Ractive instance. I think the doc needs more context about that.

There's not really any difference to Ractive. The difference to the _user_ is that you can't use a non-component instance in a regular component tag, though I suppose you could force it with something like `new Ractive({ template: '<foo />', components: { foo: Ractive } });`, but that seems quite counter-intuitive. The point of most embedded components that are registered is that they have their own template, data, and whatnot and interact with the context in some way. Anchored instances are usually more standalone and used in templates that act more as a layout.

fskreuz commented on docs/API/Instance Methods.md:
````diff
@@ -410,7 +474,7 @@ This can be used to great effect with method events and the [`@keypath`](../Conc
 ```
 
-Links can be removed using [`ractive.unlink()`](../API/Instance Methods.md#ractive.unlink()).
+Links can be removed using [`ractive.unlink()`](#unlink).
````
Nice catch. Never thought of just adding them as hashes. :D

fskreuz commented:
@Jeff17Robbins This good to go?

Jeff17Robbins commented:
I added Chris's text on the static option for observe, fixed some more
formatting, and fixed all the local anchor/bookmark links.  I didn't know
that the markdown automatic anchor for, say,

## ractive.attachChild()

is

#ractiveattachchild

(no punctuation, all lower case)

It still lacks examples for each function.

I think it is good to go.

On Thu, Mar 2, 2017 at 8:22 AM Joseph Descalzota <notifications@github.com>
wrote:

> @Jeff17Robbins <https://github.com/Jeff17Robbins> This good to go?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/ractivejs/ractivejs.github.io/pull/17#issuecomment-283651803>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACAWh2FbnVBpNHbUAqMWG8r1iWzVHVvPks5rhsKMgaJpZM4L8qhy>
> .
>

fskreuz commented:
Nice! Thanks for the contribution! 🎉
